domain/src/DomainNegotiator.php

```diff
@@ -91,6 +87,15 @@ public function setRequestDomain($httpHost, $reset = FALSE) {
     if (!empty($id)) {
       $this->setActiveDomain($domain);
     }
+    // Fallback to default domain if no match.
+    elseif ($domain = $this->domainLoader->loadDefaultDomain()) {
+      $this->moduleHandler->alter('domain_request', $domain);
+      $domain->setMatchType(DOMAIN_MATCH_NONE);
+      $id = $domain->id();
+      if (!empty($id)) {
```

Comment:
Thanks for taking time to review the PR. Indeed I didn't want to change too much the original code.

I was mainly wondering if the approach taken to resolve my issue was the good one, or if I was missing something on how `domain` works.